pageserver/src/page_cache.rs

```diff
@@ -308,7 +307,7 @@ impl Drop for PageWriteGuard<'_> {
         assert!(self.inner.key.is_some());
         if !self.valid {
             let self_key = self.inner.key.as_ref().unwrap();
-            PAGE_CACHE.get().unwrap().remove_mapping(self_key);
+            Handle::current().block_on(PAGE_CACHE.get().unwrap().remove_mapping(self_key));
```

Comment:
What guarantees we have that a Drop will never run within async context?

Reply:
I think it is possible to use `tokio::task::block_in_place` to avoid this, [playground](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=4613c012c8818489346e3fdafa8fe01d).

I will push a version that uses `block_in_place`.

Reply:
Please don't use it, we'll chat more. I should add it as denylisted, or these both.

Reply:
Made a PR to make this unneccessary: #5111

Reply:
This issue is resolved now thanks to #5129, which has been merged.